cms/static/js/views/course_video_settings.js

```diff
@@ -0,0 +1,537 @@
+                $fidelityContainer.show();
+            } else {
+                $fidelityContainer.hide();
+            }
+        },
+
+        populateLanguages: function() {
+            var self = this,
+                $languagesPreferanceContainer = self.$el.find('.transcript-languages-wrapper'),
+                $languagesContainer = self.$el.find('.languages-container');
+
+            // Clear error state if present any.
+            this.clearPreferanceErrorState($languagesPreferanceContainer);
+
+            $languagesContainer.empty();
+
+            // Show language container if provider is 3PlayMedia, else if fidelity is selected.
+            if (self.selectedProvider === '3PlayMedia' || self.selectedFidelityPlan) {
+                self.availableLanguages = self.getPlanLanguages();
+                _.each(self.activeLanguages, function(activeLanguage) {
+                    // Only add if not in the list already.
+                    if (_.indexOf(self.selectedLanguages, activeLanguage) === -1) {
+                        self.selectedLanguages.push(activeLanguage);
+                        self.addLanguage(activeLanguage);
+                    }
+                });
+                $languagesPreferanceContainer.show();
+                self.populateLanguageMenu();
+            } else {
+                self.availableLanguages = {};
+                $languagesPreferanceContainer.hide();
+            }
+        },
+
+        populateLanguageMenu: function() {
+            var availableLanguages,
+                $languageMenuEl = this.$el.find('.transcript-language-menu'),
+                $languageMenuContainerEl = this.$el.find('.transcript-language-menu-container'),
+                selectOptionEl = new Option(gettext('Select language'), '');
+
+            // Omit out selected languages from selecting again.
+            availableLanguages = _.omit(this.availableLanguages, this.selectedLanguages);
+
+            // If no available language is left, then don't even show add language box.
+            if (_.keys(availableLanguages).length) {
+                $languageMenuContainerEl.show();
+                // We need to set id due to a11y aria-labelledby
+                selectOptionEl.id = 'transcript-language-none';
+
+                HtmlUtils.setHtml(
+                    $languageMenuEl,
+                    HtmlUtils.HTML(selectOptionEl)
+                );
+
+                _.each(availableLanguages, function(value, key) {
+                    HtmlUtils.append(
+                        $languageMenuEl,
+                        HtmlUtils.HTML(new Option(value, key))
+                    );
+                });
+            } else {
+                $languageMenuContainerEl.hide();
+            }
+        },
+
+        populatePreferences: function() {
+            this.populateProvider();
+            this.populateTurnaround();
+            this.populateFidelity();
+            this.populateLanguages();
```

Comment:
Similarly, all of the methods are actually rendering the content rather that just populating the values. can we change them.